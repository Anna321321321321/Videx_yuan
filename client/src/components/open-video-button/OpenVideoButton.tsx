import React from 'react';
import { Button } from 'antd';

interface OpenVideoButtonProps {
  courseId: string;
  lessonId: string;
  start: number;
  end: number;
}

export default (props: OpenVideoButtonProps) => (
  <Button
    style={{ borderRadius: 0, float: 'right', marginTop: 3 }}
    type="primary"
    onClick={() =>
      window.open(
        `/course/${props.courseId}/lesson/${props.lessonId}?start=${
          props.start
        }&end=${props.end}`
      )
    }
  >
    Open Video
  </Button>
);
