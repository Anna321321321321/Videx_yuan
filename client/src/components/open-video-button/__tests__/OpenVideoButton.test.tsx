import React from 'react';
import { mount } from 'enzyme';
import OpenVideoButton from '..';

describe('OpenVideoButton Unit Tests', () => {
  it('When active button clicked, will open course video', () => {
    // setup
    const mockedOpen = jest.fn();
    const originalOpen = window.open;
    window.open = mockedOpen;
    const wrapper = mount(
      <OpenVideoButton
        courseId={'courseId'}
        lessonId={'lessonId'}
        start={0}
        end={100}
      />
    );

    // tests
    const button = wrapper.find('Button');
    button.simulate('click');
    expect(mockedOpen).toBeCalled();

    // cleanup
    window.open = originalOpen;

    expect(wrapper).toMatchSnapshot();
  });
});
