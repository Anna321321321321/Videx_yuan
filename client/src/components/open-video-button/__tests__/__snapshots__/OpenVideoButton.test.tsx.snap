// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpenVideoButton Unit Tests When active button clicked, will open course video 1`] = `
<Component
  courseId="courseId"
  end={100}
  lessonId="lessonId"
  start={0}
>
  <Button
    block={false}
    ghost={false}
    loading={false}
    onClick={[Function]}
    prefixCls="ant-btn"
    style={
      Object {
        "marginLeft": -105,
        "marginTop": -5,
      }
    }
  >
    <Wave>
      <button
        className="ant-btn"
        onClick={[Function]}
        style={
          Object {
            "marginLeft": -105,
            "marginTop": -5,
          }
        }
        type="button"
      >
        <span
          key=".0"
        >
          Open Video
        </span>
      </button>
    </Wave>
  </Button>
</Component>
`;
